server/storage/gdrive.go

```diff
@@ -45,13 +46,20 @@ func NewGDriveStorage(clientJSONFilepath string, localConfigPath string, basedir
 	}
 
 	// If modifying these scopes, delete your previously saved client_secret.json.
-	config, err := google.ConfigFromJSON(b, drive.DriveScope, drive.DriveMetadataScope)
-	if err != nil {
-		return nil, err
+	var httpClient *http.Client
+
+	if strings.Contains(string(b), `"type": "service_account"`) {
+		logger.Println("GDrive: using Service Account credentials")
+		httpClient = JWTConfigFromJSON(b, logger).Client(ctx)
+	} else {
+		logger.Println("GDrive: using OAuth2 credentials")
+		config, err := google.ConfigFromJSON(b, drive.DriveScope, drive.DriveMetadataScope)
+		if err != nil {
+			return nil, err
+		}
+		httpClient = getGDriveClient(ctx, config, localConfigPath, logger)
```

Comment:
let's rename `getGDriveClient` to something like `getGDriveClientFromToken`

Reply:
Done with f0963001c3e330dd2c3ef0cefe796382c58a6cb4